agent/consul/state/catalog.go

```diff
@@ -290,8 +290,18 @@ func validateRegisterRequestPeerNamesTxn(_ ReadTxn, args *structs.RegisterReques
 
 	{
 		// TODO(peering): validate the node's peering exists (skip check on restore)
-		peerName := args.PeerName
-		peerNames[peerName] = struct{}{}
+		if len(peerNames) == 1 {
```

Comment:
This should be reverted, the peer name for all objects in the request must match

Reply:
I was the problem here hehe